packages/realm-web/src/tests/utils/MockNetworkTransport.ts

```diff
@@ -67,7 +67,25 @@ export class MockNetworkTransport implements NetworkTransport {
         if (this.responses.length > 0) {
             const [response] = this.responses.splice(0, 1);
             if (response instanceof MongoDBRealmError) {
-                return Promise.reject(response);
+                return Promise.resolve({
+                    ok: false,
+                    status: response.statusCode,
+                    statusText: response.statusText,
+                    url: response.url,
+                    json: () =>
+                        Promise.resolve({
```

Comment:
Any reason to prefer `Promise.resolve({...})` over `async () => {...}`? No big deal either way, just curious if there is some reason

Reply:
No that would work.
I honestly think I've developed this habit because I've worked in codebases where I would get a lint error if I had an async function that didn't use the await keyword in its block.